### PR TITLE
Create bot for fixing overloaded import records

### PR DESCRIPTION
**Pull request by jimchamp**
Creates bot for fixing incorrectly imported BWB promise items.

### CODE REVIEW

File: promise-bot/fix_promise_items.py [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+            return
+
+        edition.local_id = updated_ids
+
+        # Fix identifiers:
+        edition.identifiers.pop("amazon", None)
+        edition.identifiers.pop("better_world_books", None)
+
+        # Fix source_records entry:
+        sku = local_id.split(":")[-1]
```
Comment (mekarpeles) [inline promise-bot/fix_promise_items.py:104]: If `local_id` is `""` are we going to get `source_ids` that look like `{source_record}:`?

Maybe fine

Comment (jimchamp) [reply promise-bot/fix_promise_items.py:117]: Throwing an exception and not writing to catalog if sku is empty.

File: promise-bot/fix_promise_items.py [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+                    break
+                edition_olid = self.extract_olid(line[:-1])  # Trim trailing newline
+
+                try:
+                    edition = self.ol.get(edition_olid)
+                    modified_fields = self.update_edition(edition)
+                    if modified_fields:
+                        self.modified += 1
+                        if not self.dry_run:
+                            edition.save(f"Modified {', '.join(modified_fields)}")
```
Comment (mekarpeles) [inline promise-bot/fix_promise_items.py:62]: lgtm ty!

**Review by mekarpeles — APPROVED** [review]

**Comment by cclauss** [comment]
Please rebase to the current `master` to get the test passing.

**Comment by cclauss** [comment]
Dear Mr. Testa,

    You requested 20K CHF to pay for our time with your architect.

    We were required to pay 20K CHF to finally get the complete description of the project.  From the contract and the binder we discovered with great surprise:
1. Alarmingly high occupancy of the building
2. True scope of the architect’s work
3. Low standard of budget items

Alarmingly high occupancy
We were surprised to discover the 42 apartments in the lower part of the building which were not present in the prospectus that we received from you.  These important elements appear to have been deliberately hidden from us.

True scope of the architect
Küchel was evidently not involved in the selection materials for the interior (kitchen, bathroom) budget so the look and feel is old-fashioned and not in line with Küchel style.  The Architect came to our meeting late and left early and dismissed of most of our requests in a way that was not customer-oriented.

Low standard of budget items
The low quality materials for the interior (kitchen, bathroom) would require substantial additional investment to upgrade most budget items.

After we received adequate information in this Friday meeting, we felt cheated.  We decided quickly.  We are very respectful and honest so we notified you of our decision the next business day.

Also, we were not informed that the costs of the notary would be required at this early stage of this process.  The edits to the draft contract were minimal and we never received an English translation.

Given your statement to me that the 20K CHF was to be used to cover the costs the Architect, we again request that you return a substantial portion of this amount to us using the bank details that Fabienne has provided.

**Comment by jimchamp** [comment]
Will do.  Should I open a new PR, or can this one be re-opened?

**Comment by jimchamp** [comment]
No worries.  Thanks @cclauss!

**Pull request merged**